### ISSUE TITLE
There is no caption extraction from types.Message when using TelegramClient.send_message with a file

### ISSUE THREAD

**Issue by GregEremeev** [LonamiWebs/Telethon#4499]
### Code that causes the issue

```python
from telethon import TelegramClient
from telethon.tl.types import Message, PeerChannel, Photo, MessageMediaPhoto


WRITE_TO_CHANNEL_NAME = NotImplemented
API_ID = NotImplemented
API_HASH = NotImplemented


async def example(client):
    message = Message(
        id=23,
        peer_id=PeerChannel(1078071132),
        date=None,
        message='message',
    )
    photo = Photo(
        dc_id=5,
        id=1358272180771727307,
        access_hash=4856299325127254003,
        file_reference=b'\x02@B\x0f\\\x00\x00\x00\x17g\x1a\x8e\xe7\xae\xe5\xd2\x7fl\xc7\x16\xc7@7\xab\x1bu\x0cr\x1f',
        sizes=[],
        date=None,
    )
    msg_media_photo = MessageMediaPhoto(photo=photo)

    await client.send_message(
        entity=WRITE_TO_CHANNEL_NAME,
        message=message,
        file=msg_media_photo,
    )

if __name__ == '__main__':
    client = TelegramClient('session_name', API_ID, API_HASH)
    client.start()
    client.loop.run_until_complete(example(client))
```

### Expected behavior

1. Run the script
2. A new post with the photo and the message successfully added
<img width="383" alt="Captura de pantalla 2024-10-24 a la(s) 3 26 30 p m" src="https://github.com/user-attachments/assets/d014da56-acce-4963-b153-34407caecb66">

### Actual behavior

The error _**TypeError: 'Message' object is not iterable**_ occurs

### Traceback

```
Traceback (most recent call last):
  File "/Users/greg/Documents/code/python/poc_repost_tg/main.py", line 97, in <module>
    client = TelegramClient('session_name', API_ID, API_HASH)
  File "/Users/greg/Documents/code/python/poc_repost_tg/.venv/lib/python3.13/site-packages/telethon/client/telegrambaseclient.py", line 328, in __init__
    self.api_id = int(api_id)
                  ~~~^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NotImplementedType'
(.venv) ➜  poc_repost_tg python main.py
Traceback (most recent call last):
  File "/Users/greg/Documents/code/python/poc_repost_tg/main.py", line 96, in <module>
    client.loop.run_until_complete(example(client))
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.13/3.13.0_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/asyncio/base_events.py", line 721, in run_until_complete
    return future.result()
           ~~~~~~~~~~~~~^^
  File "/Users/greg/Documents/code/python/poc_repost_tg/main.py", line 87, in example
    await client.send_message(
    ...<3 lines>...
    )
  File "/Users/greg/Documents/code/python/poc_repost_tg/.venv/lib/python3.13/site-packages/telethon/client/messages.py", line 831, in send_message
    return await self.send_file(
           ^^^^^^^^^^^^^^^^^^^^^
    ...<8 lines>...
    )
    ^
  File "/Users/greg/Documents/code/python/poc_repost_tg/.venv/lib/python3.13/site-packages/telethon/client/uploads.py", line 406, in send_file
    await self._parse_message_text(caption, parse_mode)
  File "/Users/greg/Documents/code/python/poc_repost_tg/.venv/lib/python3.13/site-packages/telethon/client/messageparse.py", line 87, in _parse_message_text
    message, msg_entities = parse_mode.parse(message)
                            ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/Users/greg/Documents/code/python/poc_repost_tg/.venv/lib/python3.13/site-packages/telethon/extensions/markdown.py", line 68, in parse
    message = add_surrogate(message)
  File "/Users/greg/Documents/code/python/poc_repost_tg/.venv/lib/python3.13/site-packages/telethon/helpers.py", line 44, in add_surrogate
    if (0x10000 <= ord(x) <= 0x10FFFF) else x for x in text
                                                       ^^^^
TypeError: 'Message' object is not iterable
```

### Telethon version

1.37

### Python version

3.10, 3.11, 3.12, 3.13

### Operating system (including distribution name and version)

Darwin 24.0.0 Darwin Kernel Version 24.0.0: Tue Sep 24 23:36:30 PDT 2024; root:xnu-11215.1.12~1/RELEASE_X86_64 x86_64

### Checklist

- [X] The error is in the library's code, and not in my own.
- [X] I have searched for this issue before posting it and there isn't an open duplicate.
- [X] I ran `pip install -U https://github.com/LonamiWebs/Telethon/archive/v1.zip` and triggered the bug in the latest version.

**Comment by Lonami** [comment]
Appreciate the detailed report and subsequent PR!

**Comment by GregEremeev** [comment]
The real use case where I've found this error it's when I iterated over posts from some telegram channel and tried to use the messages to repost them to another channel. Some posts have multiple media content(photos, videos, etc) and I'd like to send all related media content via `file` argument when I repost messages. In this way a reposted message will have grouped media content as it's supposed to be.

The example in the issue is shorter and more "technical".
